src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -807,10 +818,6 @@ - (void)asyncDispatchToMatterQueue:(void (^)(chip::Controller::DeviceCommissione
     });
 }
 
-@end
-
-@implementation MTRDeviceController (InternalMethods)
```

Comment:
This is because the "MTRDeviceController_Internal.h" header got included?  Ideally we would keep the separation between internal and non-internal things here....  Why was that header needed?

I guess the point is that we need `sessionTransportTypeForDevice` to be in the non-InternalMethods bits to be able to touch the `_cppCommissioner` ivar?

Reply:
The `.mm` file should be including it's own `_Internal.h` header since it's meant to be implementing the whole class, including the internal-only methods. The fact that it didn't means the signatures of the  implementation of the internal method weren't actually being checked against what was defined in the `InternalMethods` category. Also the usual pattern is to use a class extension (aka "unnamed category") for distinguishing public/private/internal parts of a class interface, instead of a named category (those are really designed for additions made from outside the class itself).